enumerable.rb

```diff
@@ -1,87 +1,163 @@
+      else
+        return my_each { |v| return true if v.is_a?(*args) } == false if args[0].instance_of?(Class)
+        return my_each { |v| return true if v.match(args[0]) } == false if args[0].instance_of?(Regexp)
 
-# p [nil, nil, nil].my_all?
-# p [nil, nil, nil].all?
-
-# p ["some", "somithimes", "something"].my_all?(/s/)
-# p ["some", "somithimes", "something"].all?(/s/)
+        my_each { |v| return true if v == args[0] } == false
+      end
+    else
+      my_each { |v| return true if yield(v) } == false
+    end
+  end
 
-# p %w[ant bear cat].my_all?(/t/)  
-# p %w[ant bear cat].all?(/t/)  
+  def my_none?(*args)
+    if !block_given?
+      if args.empty?
+        if !empty?
+          my_each do |v|
+            return false unless v.nil? && (v != false || false)
+          end == self
+        else
+          true
+        end
+      else
+        my_each { |v| return false if v.is_a?(*args) } == self if args[0].instance_of?(Class)
+        my_each { |v| return false if v.match(args[0]) } == self if args[0].instance_of?(Regexp)
+        my_each { |v| return false if v == args[0] } == self
+      end
+    else
+      my_each { |v| return false if yield(v) } == self
+    end
+  end
 
-# p %w[ant bear cat].my_all?(/a/) 
-# p %w[ant bear cat].all?(/a/) 
+  def my_count(*args)
+    count = 0
+    if block_given? == false
+      if args.empty?
+        my_each { |_v| count += 1 }
+      else
+        my_each { |v| count += 1 if v == args[0] }
+      end
+    else
+      my_each { |v| count += 1 if yield(v) }
+    end
+    count
+  end
 
-# p [1, 2i, 3.14, 5].my_all?(Numeric) 
-# p [1, 2i, 3.14, 5].all?(Numeric) 
+  def my_map(&proc)
+    return to_enum(:my_map) unless block_given?
 
-# p %w[ant bear cat].my_all? { |word| word.length >= 3 }
-# p %w[ant bear cat].all? { |word| word.length >= 3 }
+    array = []
+    to_a.my_each { |v| array << proc.call(v) }
+    array
+  end
 
-# p %w[ant bear cat].my_all? { |word| word.length >= 4 }
-# p %w[ant bear cat].all? { |word| word.length >= 4 }
+  def my_inject(*args)
+    accu = 0
+    if block_given? == false
+      accu = args.size < 2 ? to_a[0] : args[0]
+      to_a.my_each_with_index do |_v, i|
+        if args.size < 2
+          break if to_a[i + 1].nil?
+          accu = accu.send(args[0], to_a[i + 1])
+        else
+          accu = accu.send(args[1], to_a[i])
+        end
+      end
+    elsif args.empty?
+      i = 0
+      accu = to_a[0]
+      while i < to_a.length - 1 || !to_a[i + 1].nil?
+        accu = yield(accu, to_a[i + 1])
+        i += 1
+      end
+    else
+      i = 0
+      accu = args[0]
+      while i < to_a.length
+        accu = yield(accu, to_a[i])
+        i += 1
+      end
+    end
+    accu
+  end
+end
 
-# p %w[ant bear cat].my_all? 
-# p %w[ant bear cat].all? 
+def multiply_els(args)
+  args.my_inject(:*)
+end
 
-# p [ nil, true, 99].my_all?  
-# p [nil, true, 99].all?  
+p [1, 3, 4, 5].my_inject(2, &:+)
 
-# p [].my_all?
-# p [].all?
```

Comment:
- [ ] Remove debugging instructions and comments

Reply:
Thank you for the test code, it helps a lot.